server/src/com/thoughtworks/go/server/service/ArtifactsService.java

```diff
@@ -55,6 +55,11 @@
     @Autowired
     public ArtifactsService(JobResolverService jobResolverService, StageDao stageDao,
                             ArtifactsDirHolder artifactsDirHolder, ZipUtil zipUtil, SystemService systemService) {
+        this(jobResolverService, stageDao, artifactsDirHolder, zipUtil, systemService, new ArtifactDirectoryChooser());
```

Comment:
This can be fully autowired, I suppose. There's nothing special in ```ArtifactDirectoryChooser``` that I can see.